### ISSUE TITLE
data attributes are not propagated from objc_library to cc_library or cc_binary

### ISSUE THREAD

**Issue by senchangniantic** [bazelbuild/bazel#16270]
### Description of the bug:

When building a cc_binary with objc_library as dependency , the data in objc_library is not copied as runfile for the cc_binary.

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

[Archive.zip](https://github.com/bazelbuild/bazel/files/9559876/Archive.zip)

### Which operating system are you running Bazel on?

macOS 12.2.1 on m1

### What is the output of `bazel info release`?

release 5.2.0

### Have you found anything relevant by searching the web?

https://github.com/bazelbuild/bazel/issues/11425 is possibly related

**Comment by yyuting** [comment]
Hello, I'm having the same issue with release 7.1.0. Have you found a workaround? Thanks.

**Comment by jiawen** [comment]
@keith Does this have anything to do with the Objective-C transition (#16870)?

@yyuting On the basis of just a hunch, you can try `--incompatible_disable_objc_library_transition`. It will probably break numerous other things in your larger project however.

**Comment by keith** [comment]
I don't think so but I haven't tested. I assume that there's somewhere in the data collection that's only looking at certain rule types but I haven't looked. I imagine if we could find the spot it could be an easy fix

**Comment by fmeum** [comment]
Looking at `objc_library.bzl`, it doesn't seem to be doing anything with `ctx.attr.data` and instead only returns the output archives as `data_runfiles` (which also seems unnecessary).

**Comment by keith** [comment]
I imagine data has mostly been ignored there because most apple app resources are handled differently

**Comment by jiawen** [comment]
Still relevant for non-Apple platforms. On Apple platforms, there are now workarounds using an aspect that controls resource collecting.

**Comment by keith** [comment]
https://github.com/bazelbuild/rules_cc/pull/632
